timely_beliefs/beliefs/utils.py

```diff
@@ -684,12 +684,14 @@ def interpret_special_read_cases(
             timezone_to_localize_to=timezone,
         )
         if resample:
-            df = (
-                df.set_index("event_start")
-                .resample(sensor.event_resolution)
-                .mean()
-                .reset_index()
-            )
+            df = df.set_index("event_start")
+            if df.index.freq is None and len(df) > 2:
+                df.index.freq = pd.infer_freq(df.index)
+            if df.index.freq is not None and df.index.freq > sensor.event_resolution:
+                df = df.resample(sensor.event_resolution).ffill()
+            else:
```

Comment:
This else block is triggered if `df.index.freq is None and df.index.freq > sensor.event_resolution`, which is not your intention, I believe.

Better to first check `df.index.freq is not None`, then to decide if `ffill()`or `mean()` should be used?

We did not have that check in the existing code, so why do we need it now?

Reply:
> This else block is triggered if df.index.freq is None and df.index.freq > sensor.event_resolution, which is not your intention, I believe.

Your statement would fail on comparing None to a timedelta, but I'll take your comment as a worry that one might accidentally downsample when they wanted to upsample, in case the data frequency could not be determined (e.g. in case of irregularly spaced data). Maybe I could add a user warning in case the data frequency cannot be determined?

I already added a few inline comments to clarify the new logic.

Reply:
Thanks for clarifying - that seems to be the situation we don't want to happen.

I beklieve my point about first checking if we have `df.index.freq`  now is valid. If it's still `None` after line 690, shouldn't we print a warning and abort?